File: CultureLinkCRM/CultureLinkCRM.Web/Modules/ServerTypes/CrmDB/ZipCodeRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface ZipCodeRow {
    ZipCodeId?: number;
    ZipCode?: string;
    ListInd?: number;
}

export abstract class ZipCodeRow {
    static readonly idProperty = 'ZipCodeId';
    static readonly nameProperty = 'ZipCode';
    static readonly localTextPrefix = 'CrmDB.ZipCode';
    static readonly lookupKey = 'CrmDB.ZipCode';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<ZipCodeRow>('CrmDB.ZipCode') }
    static async getLookupAsync() { return getLookupAsync<ZipCodeRow>('CrmDB.ZipCode') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<ZipCodeRow>();
}

File: CultureLinkCRM/CultureLinkCRM.Web/Modules/CrmDB/ZipCode/ZipCodeRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace CultureLinkCRM.CrmDB;

[ConnectionKey("Default"), Module("CrmDB"), TableName("ZipCode")]
[DisplayName("Zip Code"), InstanceName("Zip Code")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
[LookupScript]
public sealed class ZipCodeRow : Row<ZipCodeRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Zip Code Id"), PrimaryKey, IdProperty]
    public int? ZipCodeId { get => fields.ZipCodeId[this]; set => fields.ZipCodeId[this] = value; }

    [DisplayName("Zip Code"), QuickSearch, NameProperty]
    public string ZipCode { get => fields.ZipCode[this]; set => fields.ZipCode[this] = value; }

    [DisplayName("List Ind"), Column("list_ind")]
    public int? ListInd { get => fields.ListInd[this]; set => fields.ListInd[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field ZipCodeId;
        public StringField ZipCode;
        public Int32Field ListInd;

    }
}